src/razor/src/razorLanguageServerClient.ts

```diff
@@ -223,7 +223,7 @@ export class RazorLanguageServerClient implements vscode.Disposable {
     }
 
     public async connectNamedPipe(pipeName: string): Promise<void> {
-        await this.startHandle;
+        await this.start();
```

Comment:
I'm not sure I understand this. You said that previous it could wait on `startHandle` before it had been created, due to delayed initialization, which makes sense, but this makes it look like this will actually start things up instead. Doesn't that break the delayed initialization?

This change makes sense if we know that `connectNamedPipe` is only called after something else has started the Razor bits, but if that were true then I don't see how the old code would have been problematic.

Or does `start()` have a wait inside it that waits for a file to appear in the workspace? That would make sense, but if so, perhaps a comment here mentioning that, and why its okay to call start here, would be good to add.

Reply:
From my understanding this happens anytime roslyn would ask for dynamic file information from razor: https://github.com/dotnet/vscode-csharp/blob/ec1d55dc1b5cf06cf0500d28d4c15565c523b261/src/razor/src/document/razorDocumentManager.ts#L173 

At that point we should be starting up right? Otherwise dynamic file info will never be provided. If that's not the case then I could add this to check if razor is started, add a callback for after it started

Reply:
Okay, I see it now. There is a gap where a there could be no Razor files in the VS Code workspace, but they do exist in a Roslyn project. Odd, but fair enough. I had a quick look at it seems fine if this happens to be where the language server starts.